### ISSUE TITLE
Improve/augment content tag to achieve uniqueness

### ISSUE THREAD

**Issue by perolavsvendsen** [equinor/fmu-dataio#123]
Offline discussion: How to get uniqueness given current schema (version 0.8.0).

Include `data.content_attributes`:
```
data:
  content: <str> <-- Standardized, whitelist in fmu-dataio. So not very granular.
  content_attributes:  <-- list of dicts, not explicitly indexed in Elastic
    tag: <str> <-- always present, not standardized content, always a string
    xx: <yy>
```

(In examples, a `properties` tag is included, which is a list. Not used currently.)
Suggest to reverse the current `seismic` (and other similar), use `data.content_attributes` instead.
User must be able to include custom attributes under `data.content_attributes`.

Fundamental requirement here is to be able to find the right data, using metadata alone. (Technically, we _can_ use the `file.relative_path`.)

Previous versions on JS:
```
content: mycontent
mycontent:
  origin: mycontent_origin
```

@jcrivenaes @jsorb

**Comment by perolavsvendsen** [comment]
Increment to 0.8.1

**Comment by perolavsvendsen** [comment]
Give warning if `content_attributes` are missing for now, then at some point make mandatory.

**Comment by perolavsvendsen** [comment]
`content_attributes` is possibly a use case for `flattened objects` in Elasticsearch.

**Comment by perolavsvendsen** [comment]
Discussion 2022-02-28, add `data.tag` to provide uniqueness.

`tagname` is already (optional) input argument. Perhaps we make this mandatory (in the future), and record it in the metadata as `data.tag`. 

The purpose is to ensure uniqueness of data of identical class/content. Currently, the only differentiator will be `file.relative_path` or `fmu.workflow` (but the latter is sketchy as well).

**Comment by perolavsvendsen** [comment]
tagname was included in #194

**Comment by perolavsvendsen** [comment]
Old issue, closing this
